Python/pylifecycle.c

```diff
@@ -1210,6 +1210,14 @@ finalize_interp_clear(PyThreadState *tstate)
 {
     int is_main_interp = _Py_IsMainInterpreter(tstate);
 
+    /* bpo-36854: Explicitly clear the codec registry
+       and trigger a GC collection */
+    PyInterpreterState *interp = tstate->interp;
+    Py_CLEAR(interp->codec_search_path);
+    Py_CLEAR(interp->codec_search_cache);
+    Py_CLEAR(interp->codec_error_registry);
+    _PyGC_CollectNoFail();
+
     /* Clear interpreter state and all thread states */
     PyInterpreterState_Clear(tstate->interp);
```

Comment:
IMHO, Can it be replaced by the newly declared variable?

`PyInterpreterState_Clear(interp);`

Reply:
Oh, right, interp could be used. I hesitate between adding "interp" variables or using "tstate->interp" :-) I may change this in my next PR, I already merged this once before I read your comment, sorry.

Reply:
Thank you for the reply :)